src/goto-programs/goto_convert_side_effect.cpp

```diff
@@ -315,6 +317,7 @@ void goto_convertt::remove_post(
   {
     exprt tmp = op;
     make_temp_symbol(tmp, "post", dest, mode);
+    tmp.set(ID_tmp_post_origin, exprt(expr));
```

Comment:
Did the idea of setting `pretty_name` of the symbol introduced by `make_temp_symbol` not work?

Reply:
No. The `pretty_name` of the symbol introduced by `make_temp_symbol` is not printed in the report and the trace. They instead print `from_expr(e)` where `e` is the `tmp_post` variable with type `symbol_exprt`. 
I think although we can add a special case into `expr2c` to print `pretty_name`, it suffers the same issue as Daniel  pointed out.  Printing `tmp_post` as `*ptr++` is very confusing that we cannot distinguish if it is the 'real' `*ptr++`, or it is `tmp_post`'s pretty name.